### ISSUE TITLE
We can also implement:

### ISSUE THREAD

**Issue by QGarchery** [morpho-dao/morpho-data-structures#64]
We can also implement:
- #38 
- #36

_Originally posted by @QGarchery in https://github.com/morpho-dao/morpho-data-structures/issues/23#issuecomment-1195113603_

**Comment by MerlinEgalite** [comment]
Can we close? @QGarchery

**Comment by makcandrov** [comment]
I'm not in favor of resolving #38 because it will increase gas consumption in general cases to save gas in a rare case where two accounts have the same value.

Regarding #36, it is correct that we can make the heap more usable by adding/replacing some functions. Which functions should be added?
- A function `replace` instead of `increase` and `decrease`.
- A function `pop` that removes the first account of the heap.
- A function `popAndInsert` that removes the first account and inserts another account.

I'm in favor of the first two functions, but not the third.

**Comment by QGarchery** [comment]
> I'm not in favor of resolving https://github.com/morpho-dao/morpho-data-structures/issues/38 because it will increase gas consumption in general cases to save gas in a rare case where two accounts have the same value.

I don't think we can assume that the rare case is when two accounts have the same value. I understand that there are use cases where indeed accounts should rarely have the same value (and that's why it's not implemented for Morpho, see [this comment](https://github.com/morpho-dao/morpho-data-structures/issues/38#issuecomment-1181806615)). But this implementation is meant as a general purpose heap, so we can make no assumption about how it's used.

I like the update function: it seems to be lightweight in terms of implementation and would be useful in a lot of cases
